src/doc/trpl/guessing-game.md

```diff
@@ -0,0 +1,991 @@
+we didn’t include a return type, it’s assumed to be `()`, an empty
+[tuple][tuples].
+
+[tuples]: primitive-types.html#tuples
+
+```rust,ignore
+    println!("Guess the number!");
+
+    println!("Please input your guess.");
+```
+
+We previously learned that `println!()` is a [macro][macros] that
+prints a [string][strings] to the screen.
+
+[macros]: macros.html
+[strings]: strings.html
+
+```rust,ignore
+    let mut guess = String::new();
+```
+
+Now we’re getting interesting! There’s a lot going on in this little line. The first thing to notice is that this is a [let statement][let], which is used to create ‘variable bindings’. They take this form:
+
+```rust,ignore
+let foo = bar;
+```
+
+[let]: variable-bindings.html
+
+This will create a new binding named `foo`, and bind it to the value `bar`. In
+many languages, this is called a ‘variable’, but Rust’s variable bindings have
+a few tricks up their sleeves.
+
+For example, they’re [immutable][immutable] by default. That’s why our example
+uses `mut`: it makes a binding mutable, rather than immutable. `let` doesn’t
+take a name on the left hand side, it actually accepts a
+‘[pattern][patterns]’. We’ll use patterns more later. It’s easy enough
+to use for now:
+
+```
+let foo = 5; // immutable.
+let mut bar = 5; // mutable
+```
+
+[immutable]: mutability.html
+[patterns]: patterns.html
+
+Oh, and `//` will start a comment, until the end of the line. Rust ignores
+everything in [comments][comments].
+
+[comments]: comments.html
+
+So now we know that `let mut guess` will introduce a mutable binding named
+`guess`, but we have to look at the other side of the `=` for what it’s
+bound to: `String::new()`.
+
+`String` is a string type, provided by the standard library. A
+[`String`][string] is a growable, UTF-8 encoded bit of text.
+
+[string]: ../std/string/struct.String.html
+
+The `::new()` syntax is uses `::` because this is an ‘associated function’ of
+a particular type. That is to say, it’s associated with `String` itself,
+rather than a particular instance of a `String`. Some languages call this a
+‘static method’.
+
+This function is named `new()`, because it creates a new, empty `String`.
+You’ll find a `new()` function on many types, as it’s a common name for making
+a new value of some kind.
+
+Let’s move forward:
+
+```rust,ignore
+    io::stdin().read_line(&mut guess)
+        .ok()
+        .expect("Failed to read line");
+```
+
+That’s a lot more! Let’s go bit-by-bit. The first line has two parts. Here’s
+the first:
+
+```rust,ignore
+io::stdin()
+```
+
+Remember how we `use`d `std::io` on the first line of the program? We’re now
+calling an associated function on it. If we didn’t `use std::io`, we could
+have written this line as `std::io::stdin()`.
+
+This particular function returns a handle to the standard input for your
+terminal. More specifically, a [std::io::Stdin][iostdin].
+
+[stdin]: http://doc.rust-lang.org/nightly/std/io/struct.Stdin.html
```

Comment:
This reference and the above don't match (`iostdin` vs `stdin`) and this should use a relative path (`../std/io/struct.Stdin.html`).